src/cmds/qdel.c

```diff
@@ -334,30 +346,38 @@ char **envp;
 		 *   "nomailforcedeletehist" -- force delete history of a job without sending mail.
 		 */
 		mails = dfltmail ? dfltmail : 1000;
-		if (num_deleted >= mails && !mails_suppressed) {
-			mails_suppressed = TRUE;
-			/* current warg1 "nomail" should be at start */
-			strcat(warg1, warg);
-			pbs_strncpy(warg, warg1, sizeof(warg));
-		}
-
-		p_delstatus = pbs_deljoblist(connect, svr_itr->jobids, warg);
+		numofjobs = (mails <= svr_itr->job_ct) ? mails : svr_itr->job_ct;
 		
-		while (p_delstatus != NULL) {
-			if (p_delstatus->code == PBSE_UNKJOBID) {
-				if (locate_job(p_delstatus->name, server_out, rmt_server))
-					add_jid_to_list(rmt_server, rmt_server, &svr_itr);
-				continue;
+		for (i = 0; i < svr_itr->job_ct; i = i + numofjobs) {
+			if (mails_suppressed)
+				numofjobs = ((i + NUM_JOBS_IN_A_BATCH) <= svr_itr->job_ct) ? NUM_JOBS_IN_A_BATCH : (svr_itr->job_ct - i);
```

Comment:
why i + NUM_JOBS_IN_A_BATCH ? this will mean that the first pbs_deljoblist will delete 10k, the next one 20k, and so on, right?

Reply:
This condition "(i + NUM_JOBS_IN_A_BATCH) <= svr_itr->job_ct)" equals to "current job index + 10k" is less then total count. If this condition succeeds means, qdel has enough jobs further to delete, then assigning 10k to "numofjobs". So in each iteration, it will be 10k always.

Reply:
ok yes, true .. but why are you checking i + NUM_JOBS_IN_A_BATCH ? after the first iteration, i will be 10k, so you'll check whether 20k is less than svr_itr->job_ct ? why not just check if NUM_JOBS_IN_A_BATCH <= svr_itr->job_ct ?

Reply:
thanks for explaining offline that 'job_ct' is the non-changing total number of jobs. Then this is fine, but please rename job_ct to something like 'total_jobs' to avoid confusion.

Reply:
Both the conditional operators does the same thing. You only have to decide the batch size in the if condition.
```
			if (!mails_suppressed)
				batch_size = mail_count;
			numofjobs = ((i + batch_size) <= svr_itr->job_ct) ? batch_size : (svr_itr->job_ct - I);
```